vendor/bat-native-ads/include/bat/ads/confirmation_type.h

```diff
@@ -6,19 +6,34 @@
 #ifndef BAT_ADS_CONFIRMATION_TYPE_H_
 #define BAT_ADS_CONFIRMATION_TYPE_H_
 
+#include <string>
+
 namespace ads {
 
-static char kConfirmationTypeClick[] = "click";
-static char kConfirmationTypeDismiss[] = "dismiss";
-static char kConfirmationTypeView[] = "view";
-static char kConfirmationTypeLanded[] = "landed";
-
-enum class ConfirmationType {
-  UNKNOWN,
-  CLICK,
-  DISMISS,
-  VIEW,
-  LANDED
+class ConfirmationType {
+ public:
+  enum Value : int {
+    UNKNOWN,
+    CLICK,
+    DISMISS,
+    VIEW,
+    LANDED
+  };
+
+  ConfirmationType() = default;
+  constexpr ConfirmationType(const Value value) : value_(value) {}
```

Comment:
lint error here, Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]

Reply:
as we discussed in slack, I'm fine with NOLINT and a comment stating that we allow implicit conversion from enum value to this wrapper class, similar as https://cs.chromium.org/chromium/src/v8/src/wasm/value-type.h?rcl=88a0c6d607cf023a5776193fe1786be0ca9c0767&l=64-65.

Reply:
👍 added NOLINT and comment as above, thanks